userspace/sysdig/sysdig.cpp

```diff
@@ -1268,18 +1317,61 @@ sysdig_init_res sysdig_init(int argc, char **argv)
 				}
 			}
 
-			if(is_filter_display)
-			{
-				sinsp_filter_compiler compiler(inspector, filter);
-				display_filter = compiler.compile();
-			}
 #else
 			fprintf(stderr, "filtering not compiled.\n");
 			res.m_res = EXIT_FAILURE;
 			goto exit;
 #endif
 		}
 
+#ifdef HAS_FILTERING
+		if (command.size()) {
+			if(infiles.size() != 0)
+			{
+				fprintf(stderr, "the --exec flag cannot be used with saved captures.\n");
+				res.m_res = EXIT_FAILURE;
+				goto exit;
+			}
+
+			if(signal(SIGCHLD, SIG_IGN) == SIG_ERR)
+			{
+				fprintf(stderr, "An error occurred while setting SIGCHLD signal handler.\n");
+				res.m_res = EXIT_FAILURE;
+				goto exit;
+			}
+
+			command_pid = fork();
+			switch(command_pid) {
+				case 0: // child
+					raise(SIGSTOP);
+					execl("/bin/sh", "/bin/sh", "-c", command.c_str(), 0);
```

Comment:
Why is the command executed with `/bin/sh -c` and not directly?

Reply:
Because all positional parameters are used to build the filter so we only have one to pass to --exec. I talked a bit about this patch with Brett yesterday and it still needs some work before considering a merge. I hope to find some better UI that will allow straight exec as well.

I'll have to see if e.g. `sysdig evt.type=open --exec cat /etc/passwd` will be easily doable (all positional args after --exec count towards the argv of the command). How does that sound?

Reply:
Not sure I understand... Are you using `/bin/sh -c` so so you can do `cat /etc/passwd` instead of `/bin/cat /etc/passwd`?

Reply:
No, I'm using the shell because with the current command line I can only get a single parameter to --exec (i.e. `--exec "cat /etc/passwd"` works, while `--exec cat /etc/passwd` makes `/etc/passwd` a part of the filter). So the detour through the shell is required* to pass arguments to the final binary (`execvp("cat /etc/passwd")` won't work).

Still, I don't like this at all, so I'll try to improve that and avoid the shell.

*well, we could split the parameters ourselves but that seems even worse